src/lib/DataCollector/TwigComponentCollector.php

```diff
@@ -20,9 +21,14 @@ final class TwigComponentCollector extends DataCollector
     /** @var array<int, mixed> */
     private array $availableGroups = [];
 
-    public function addRenderedComponent(string $group, string $name): void
+    public function addRenderedComponent(RenderSingleEvent $event): void
```

Comment:
I am not sure it is okay to make the collector aware it is working with events. I'd prefer using explicit arguments for each part of collected data due to this.

If you want to keep Component data in the profiler, you can use `DataCollector::cloneVar()` method.